presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionInterpreter.java

```diff
@@ -444,7 +444,11 @@ else if (!javaType.isPrimitive()) {
         @Override
         protected Object visitIdentifier(Identifier node, Object context)
         {
-            return node;
+            // Identifier only exists before planning.
+            // ExpressionInterpreter should only be invoked after planning.
```

Comment:
Is this true? I thought the interpreter is invoked during planning for constant folding, etc.

Reply:
No, it's not true. See "However" 2 lines down.